sdk/jobs/pipelines/3a_pipeline_with_spark/spark-in-pipeline.ipynb

```diff
@@ -0,0 +1,119 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# import required libraries\n",
+    "from azure.identity import DefaultAzureCredential, InteractiveBrowserCredential\n",
+    "from azure.ml import MLClient, Input, Output\n",
+    "from azure.ml import command, spark\n",
+    "from azure.ml.dsl import pipeline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    credential = DefaultAzureCredential()\n",
+    "    # Check if given credential can get token successfully.\n",
+    "    credential.get_token(\"https://management.azure.com/.default\")\n",
+    "except Exception as ex:\n",
+    "    # Fall back to InteractiveBrowserCredential in case DefaultAzureCredential not work\n",
+    "    credential = InteractiveBrowserCredential()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get a handle to workspace\n",
+    "ml_client = MLClient.from_config(credential=credential)\n",
+    "\n",
+    "# Retrieve an already attached Azure Machine Learning Compute.\n",
+    "synapse_compute = \"synapse-compute\"\n",
+    "print(ml_client.compute.get(synapse_compute))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define pipeline\n",
+    "@pipeline(description=\"Spark pipeline job\",)\n",
+    "def spark_pipeline_job(input_data):\n",
+    "    # define the spark task\n",
+    "    spark_node = spark(\n",
+    "        code=\"./src\",\n",
+    "        entry=\"program.py\",\n",
+    "        py_files=['my_python.py'],\n",
+    "        jars=['my_java.jar'],\n",
+    "        files=['my_file.txt'],\n",
+    "        archives=['my_archive.tar'],\n",
+    "        conf=dict({\n",
+    "            \"spark.driver.cores\": 1,\n",
+    "            \"spark.driver.memory\": \"2g\",\n",
+    "            \"spark.executor.cores\": 2,\n",
+    "            \"spark.exeutor.memory\": \"2g\",\n",
+    "            \"spark.executor.instances\": 4\n",
+    "        }),\n",
+    "        environment='myenv:v1',\n",
+    "        # environment={\"conda_file\": \"./conda.config\"},\n",
+    "        inputs=dict(input_folder=input_data),\n",
```

Comment:
we seem to miss the args parameter which binds the inputs & outputs

Reply:
yep will do